src/raft.rs

```diff
@@ -1993,16 +1983,24 @@ impl<T: Storage> Raft<T> {
     fn check_quorum_active(&mut self) -> bool {
         let self_id = self.id;
         let mut act = 0;
-        for (&id, pr) in self.mut_prs().iter_mut() {
-            if id == self_id {
-                act += 1;
-                continue;
+
+        self.prs = if let Some(mut prs) = self.prs.take() {
+            for (&id, pr) in prs.voters_mut() {
+                if id == self_id {
+                    act += 1;
+                    continue;
+                }
+                if pr.recent_active { act += 1; }
+                pr.recent_active = false;
             }
-            if !pr.is_learner && pr.recent_active {
-                act += 1;
+            for (&_id, pr) in prs.learners_mut() {
```

Comment:
Is it not necessary? @BusyJay .

Reply:
Not sure. But it's better keep the old behavior.